bytecomp/bytegen.ml

```diff
@@ -209,6 +217,12 @@ let rec check_recordwith_updates id e =
 let rec size_of_lambda env = function
   | Lvar id ->
       begin try Ident.find_same id env with Not_found -> RHS_nonrec end
+  | Lconst _ ->
+      (* This is a constant, so obviously not recursive. But Rec_check might
+         have treated it as Static. We rely on the fact that the compilation
+         of non-recursive values (RHS_nonrec) already handles that case
+         correctly, and return RHS_nonrec. *)
+      RHS_nonrec
```

Comment:
This is a different bugfix, right? I would rather have it in a separate commit with its own testsuite change.

Reply:
It's not a bugfix. It's the same behaviour as before the PR, but I think this case deserved its own comment, so I split it out.